src/Avalonia.Controls/ProgressBar.cs

```diff
@@ -216,8 +218,15 @@ protected override void OnPropertyChanged(AvaloniaPropertyChangedEventArgs chang
         /// <inheritdoc/>
         protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
         {
+            // dispose any previous track size listener
+            _trackSizeChangedListener?.Dispose();
```

Comment:
I think it would be better to subscribe on Attached and unsubscribe on Detached from the visual tree. This way no active subscriptions will be alive when control is created, but not attached.

Reply:
Is it safe to do? I mean will ApplyTemplate always be called before attached to visual tree?

Reply:
Yeah, it might be a problem